arex-instrumentation/dynamic/arex-dynamic-common/src/main/java/io/arex/inst/dynamic/common/DynamicClassExtractor.java

```diff
@@ -189,29 +193,60 @@ String buildMethodKey(Method method, Object[] args, String keyExpression) {
-     * resulting in inability to accurately match data. Use className to represent the time value.
+     * resulting in inability to accurately match data. And in order to be compatible with previously recorded data,
+     * the second time is cleared to zero.
+     * ex: 2023-01-01 12:12:01.123 -> 2023-01-01 12:12:00.000
      */
     private Object[] normalizeArgs(Object[] args) {
         if (ArrayUtils.isEmpty(args)) {
             return args;
         }
         Object[] normalizedArgs = new Object[args.length];
         for (int i = 0; i < args.length; i++) {
-            if (isTime(args[i])) {
-                normalizedArgs[i] = args[i].getClass().getName();
-            } else {
-                normalizedArgs[i] = args[i];
-            }
+            normalizedArgs[i] = removeTimeSecond(args[i]);
         }
         return normalizedArgs;
     }
 
-    private boolean isTime(Object arg) {
+    private Object removeTimeSecond(Object arg) {
         if (arg == null) {
-            return false;
+            return null;
+        }
+        if (arg instanceof LocalDateTime) {
+            LocalDateTime localDateTime = (LocalDateTime) arg;
+            return LocalDateTime.of(localDateTime.getYear(), localDateTime.getMonth(), localDateTime.getDayOfMonth(),
+                    localDateTime.getHour(), localDateTime.getMinute(), 0);
+        }
+        if (arg instanceof Calendar) {
+            Calendar calendar = (Calendar) arg;
+            Calendar copyCalendar = Calendar.getInstance(calendar.getTimeZone());
+            copyCalendar.setTimeInMillis(calendar.getTimeInMillis());
+            copyCalendar.set(Calendar.SECOND, 0);
+            copyCalendar.set(Calendar.MILLISECOND, 0);
+            return copyCalendar;
+        }
+        if (arg instanceof Date) {
+            Date date = (Date) arg;
+            Calendar calendar = Calendar.getInstance();
+            calendar.setTime(date);
+            calendar.set(Calendar.SECOND, 0);
+            calendar.set(Calendar.MILLISECOND, 0);
+            return calendar.getTime();
+        }
+        if (arg instanceof LocalTime) {
+            LocalTime localTime = (LocalTime) arg;
+            return LocalTime.of(localTime.getHour(), localTime.getMinute(), 0);
         }
         String className = arg.getClass().getName();
-        return arg instanceof Calendar || arg instanceof Date || className.startsWith("java.time") || className.startsWith("org.joda.time");
+        if (JODA_LOCAL_DATE_TIME.equals(className)) {
+            org.joda.time.LocalDateTime jodaLocalDateTime = (org.joda.time.LocalDateTime) arg;
+            return jodaLocalDateTime.withSecondOfMinute(0).withMillisOfSecond(0);
+        }
+        if (JODA_LOCAL_TIME.equals(className)) {
+            org.joda.time.LocalTime jodaLocalTime = (org.joda.time.LocalTime) arg;
+            return jodaLocalTime.withSecondOfMinute(0).withMillisOfSecond(0);
+        }
+        return arg;
```

Comment:
No need to renew the time instance, just format datetime and replace the seconds/milliseconds.